pkg/config/derive.go

```diff
@@ -0,0 +1,31 @@
+package config
+
+import (
+	"strings"
+
+	acsapi "github.com/openshift/openshift-azure/pkg/api"
+)
+
+func DerivedSystemReserved(cs *acsapi.OpenShiftManagedCluster, role acsapi.AgentPoolProfileRole) string {
+	for _, pool := range cs.Properties.AgentPoolProfiles {
```

Comment:
@kargakis more code that suggests AgentPoolProfiles should be a map internally?
@mjudeikis should have a single `GetAPP(cs *acsapi.OpenShiftManagedCluster, role acsapi.AgentPoolProfileRole) (*acsapi.AgentPoolProfile, error)` function in our codebase somewhere?

Reply:
> @kargakis more code that suggests AgentPoolProfiles should be a map internally?

Or create some boillerplate code that can abstract this from elsewhere? It should be a single helper that reads a []AgentPoolProfile and a role and returns a *AgentPoolProfile, right? Simpler than diverging between internal and external. Not simpler than adding a map in the API but saner I think.